src/views/MenuBar.tsx

```diff
@@ -59,11 +60,13 @@ const MenuBar = ({
   const handleFileChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
     const { files } = e.target;
     if (!files) return;
+    setFileInput(files[0]);
```

Comment:
I think since the goal of using `setFileInput` here is to trigger a re-render, making it a boolean like 
```tsx
const [isUploading, setIsUploading] = useState<boolean>(false);
```
might perhaps be better to avoid storing unnecessarily large files (if uploaded) in memory.